### PR TITLE
Map doesn't save old state

### LINKED ISSUE #36: Don't save state between games
## Overview
When more than one game is played, the state of the game map is sometimes saved. This behavior doesn't seem to be consistent.

### How this was discovered
- Mid 2014 Macbook Pro, OS X Catalina 10.15.5
- `make run` on master branch
- Without exiting the game: play one round on easy mode, then one round on hard mode, then multiple rounds on easy mode. Every few games the game state is saved between rounds.

### PR DESCRIPTION
**Pull request by drofp**
Fixes #36 

The main fix is initializing `game_map` to an empty string, so it always starts with an empty initial state (rather than using whatever was previously on the top of stack memory).

I actually haven't fully verified it, but I'm pretty confident this fix is solid after running it a few times. If you really want to verify, feel free to try this the steps below. It might be tricky figuring out the timing and how to print this value properly. If you have a C++ debugger setup, feel free to give that a shot.
- checkout master. print out the value of `game_map` in `game.h` before each run.
  - **check if this value is the same as the previous game**
- checkout save-state. repeat procedure
- this value should be the same sometimes on master, and never on save-state

### CODE REVIEW

File: src/game.cc [hunk 1]
```diff
@@ -94,35 +97,35 @@ void Game::PrintInfoFrame()
 }
 
 void Game::PrintPrepScreen()
-{  
+{
   char* mode;
   const char* kStartArt = R"ronmak(
```
Comment (evanugarte) [inline src/game.cc:99]: image

File: src/game.cc [hunk 2]
```diff
@@ -209,16 +212,16 @@ void Game::UpdateHighScore()
...
   *prev_score = curr_score;
 
   wclear(info_win);
   mvwprintw(info_win, 7, 0, "high score: %d", curr_high_score);
   mvwprintw(info_win, 8, 0, "difficulty_map difficulty: %d", difficulty_map->at(curr_difficulty));
-  
+
   wrefresh(info_win);
-  
+
```
Comment (evanugarte) [inline src/game.cc:221]: nice job getting rid of the whitespaces

Comment (drofp) [reply src/game.cc:221]: This was my VScode [Trailing Spaces plugin](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces) just automatically doing it for me. Was too lazy to turn it off just for this change lol. I wish I had this plugin when we were first developing this.

File: src/game.cc (hunk 1 above)
Comment (drofp) [reply src/game.cc:99]: The man himself. I have both him and this project to thank as part of getting me into grad school

**Comment by drofp** [comment]
also it seems like a bonus side effect of this change is that the falling blocks (the three stars -> ***) will not overlap with the walls. sometimes before this change, we would see a 3-star falling block lodged into a wall (so 1 star or 2 stars show popping out of a wall).

not exactly sure why this happens, but I'm guessing its some off-by-one with how the repaint happens on consecutive games

**Review by evanugarte — APPROVED** [review]

**Pull request merged**